File: PW7-Notifications & BroadcastReceivers/Notification/app/src/main/java/com/example/notification/MainActivity.java
package com.example.notification;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.NotificationCompat;

import android.app.Notification;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.RemoteViews;

public class MainActivity extends AppCompatActivity {

    private static final int MY_NOTIFICATION_ID = 1;

    private int mNotifCount;

    private final CharSequence contentTitle = "New Notification";
    private final CharSequence contentText = "You have a new message Check it Now!";

    private Intent mNotificationIntent;
    private PendingIntent mContentIntent;

    private Uri soundURI = Uri
            .parse("android.resource://com.example.notification/"
                    + R.raw.notifsound);
    private long[] mVibratePattern = { 0, 200, 200, 300 };




    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mNotificationIntent = new Intent(getApplicationContext(),NotificationSubActivity.class);
        mContentIntent = PendingIntent.getActivity(getApplicationContext(), 0,
                mNotificationIntent, Intent.FLAG_ACTIVITY_NEW_TASK);

        Button notif= findViewById(R.id.notify);

        notif.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Notification.Builder nBuilder = new Notification.Builder(
                        getApplicationContext())
                        .setSmallIcon(R.drawable.msg)
                        .setAutoCancel(true)
                        .setContentTitle(contentTitle)
                        .setContentText(
                                contentText + " (" + ++mNotifCount + ")")
                        .setContentIntent(mContentIntent)
                        .setSound(soundURI)
                        .setVibrate(mVibratePattern);

                NotificationManager mNotificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
                mNotificationManager.notify(MY_NOTIFICATION_ID,nBuilder.build());

            }
        });
    }
}

File: PW3/app/src/main/java/com/example/tp3/MainActivity.kt
package com.example.tp3

import android.app.Activity
import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if(savedInstanceState==null){
            var msg =intent.getStringExtra("msg")
            receivedMsg.setText(msg)
        }
        explicite.setOnClickListener{
            val intent= Intent(this@MainActivity, ExpliciteActivity::class.java)
            startActivity(intent)

        }
        implicite.setOnClickListener{
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse("https://www.google.com")))

        }


    }


}
